popup.js

```diff
@@ -0,0 +1,115 @@
+$(function() {
+  var checkbox = $("#switch-1");
+  var hidden = $("#hidden_fields");
+  var populate = $("#populate");
+  hidden.hide();
+  checkbox.change(function() {
+    if (checkbox.is(":checked")) {
+      hidden.show();
+      populate.val("Dude, this input got populated!");
+    } else {
+      hidden.hide();
+    }
+  });
+
+  var series = {
+    monthDataSeries1: {
+      bandwidthEfficiency: [88, 42, 53, 81, 68, 74],
+      time: ["3 Min", "2.5 Min", "2 Min", "1.5 Min", "1 Min", "0.5 Min"]
+    }
+  };
+
+  var options = {
+    colors: ["#F44336", "#E91E63", "#9C27B0"],
+    chart: {
+      height: 350,
+      type: "area",
+      zoom: {
+        enabled: false
+      }
+    },
+    dataLabels: {
+      enabled: false
+    },
+    stroke: {
+      curve: "straight"
+    },
+    series: [
+      {
+        name: "bandwidth efficiency (In %)",
+        data: series.monthDataSeries1.bandwidthEfficiency
+      }
+    ],
+    title: {
+      align: "left"
+    },
+    subtitle: {
+      align: "left"
+    },
+    labels: series.monthDataSeries1.time,
+    xaxis: {
+      type: "string"
+    },
+    yaxis: {
+      opposite: true
+    },
+    legend: {
+      horizontalAlign: "left"
+    }
+  };
+
+  var chart = new ApexCharts(document.querySelector("#chart"), options);
+
+  chart.render();
+});
+
+document.addEventListener("DOMContentLoaded", function() {
+  var download_btn = document.getElementById("download_btn");
+     download_btn.addEventListener("click",addDownload);
+  });
+
+function addDownload() {
+  var downurl = document.getElementsByClassName("urlText")[0].value;
+
+  if (downurl === "" || downurl === undefined || !linkvalidator(downurl)) {
+    var n = noty({
+  text: "Please Enter a valid URL",
+  layout: "bottomRight",
+  animation: {
+    closeWith: ["click"],
+    open: "animated bounceInUp",
+    close: "animated bounceOutRight"
+  }
```

Comment:
please lint and format your code. Maintaining good code from starting is good practice.

Reply:
Let's go with 1-tab spacing instead of 2 spaces